src/3d/framegraph/qgspostprocessingrenderview.h

```diff
@@ -0,0 +1,97 @@
+  Date                 : May 2025
+  Copyright            : (C) 2025 by Benoit De Mezzo and (C) 2020 by Belgacem Nedjima
+  Email                : benoit dot de dot mezzo at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSPOSTPROCESSINGRENDERVIEW_H
+#define QGSPOSTPROCESSINGRENDERVIEW_H
+
+#include "qgsabstractrenderview.h"
+#include "qgspostprocessingentity.h"
+
+namespace Qt3DRender
+{
+  class QTexture2D;
+  class QRenderTargetSelector;
+  class QClearBuffers;
+  class QRenderStateSet;
+  class QRenderCapture;
+} //namespace Qt3DRender
+
+#define SIP_NO_FILE
+
+/**
+ * \ingroup qgis_3d
+ * \brief Container class that holds different objects related to postprocessing rendering
+ *
+ * \note Not available in Python bindings
+ *
+ * The postprocessing buffer render pass is made to copy the postprocessing buffer into
+ * an RGB texture that can be captured into a QImage and sent to the CPU for
+ * calculating real 3D points from mouse coordinates (for zoom, rotation, drag..)
+ *
+ * \since QGIS 3.44
+ */
+class QgsPostprocessingRenderView : public QgsAbstractRenderView
+{
+  public:
+    //! Default constructor
+    QgsPostprocessingRenderView( const QString &viewName,                     //
+                                 QgsShadowRenderView &shadowRenderView,       //
+                                 QgsForwardRenderView &forwardRenderView,     //
+                                 QgsAmbientOcclusionRenderView &aoRenderView, //
+                                 QSize mSize,                                 //
+                                 Qt3DCore::QEntity *rootSceneEntity );
+
+    virtual void updateWindowResize( int width, int height ) override;
+
+    //! Returns the render capture object used to take an image of the postprocessing buffer of the scene
+    Qt3DRender::QRenderCapture *renderCapture() const;
+
+    //! Returns the QT3D entity used to do the rendering
+    QgsPostprocessingEntity *entity() const;
+
+    /**
+     * Sets whether it will be possible to render to an image
+     */
+    void setOffScreenRenderCaptureEnabled( bool enabled );
+
+    //! Returns the top node of all subpasses
+    QVector<Qt3DRender::QFrameGraphNode *> subPasses() const;
+
+    /**
+     * Updates the subpasses with the new \a topNodes
+     */
+    void setSubPasses( QVector<Qt3DRender::QFrameGraphNode *> topNodes );
```

Comment:
I don't really like the idea of sub-passes of the post-processing render view. The post-processing is really just one pass. The pass to enable captures of the final render should go to a separate render view.